### ISSUE TITLE
Server tps drops

### ISSUE THREAD

**Issue by ApolloMCC** [Funwayguy/BetterQuesting#598]
My tps are all over the place at random, seems like it's BQ mod, last time I had this issue when i reported it,  it was builders wand changing the inventories too fast. 
Since then building wands are banned. Here's the most recent spark. https://pasteboard.co/IdLLlnX.png

**Comment by ApolloMCC** [comment]
Could this be related that my server has 1.5k players? So 1.5k quest progression files lag the server? 
Seems like a database issue to me.

**Comment by Funwayguy** [comment]
1.5k players... oof. Didn't think people were running BQ servers of such scale yet.

yea this one is on me. I haven't rewitten the networking code to properly deal with server scaling to this degree.

It'll take some to fix this so I don't have an ETA but I'll work something out in the mean time to help.

**Comment by ApolloMCC** [comment]
If I delete some quest files from inactive players do you think that would help?
If so, where do i find those files and what do i need to delete?

**Comment by Funwayguy** [comment]
No that wouldn't fix it. The problem has more to do with me currently sending the entire progress database to every player everytime it needs updating (wasn't a problem until packs started approaching 1K quests on servers). It's a matter of sync efficiency rather than the actual data stored. Even with a few 100 players this issue is still quite noticeable with players on weak connections effectively getting DoSed off the server once it gets too large or has too many quests to sync.

The progress database is also a bulk collection of all player data so you'd need to manually delete a whole bunch of entries without breaking the formatting to trim the data of old players. It does give me an idea for doing it via command though I doubt it would make any improvement to this

**Comment by Spontini** [comment]
I dont want to be pushy, but this was reported again 2 months ago, my server is suffering from this too. Please a  temporary fix for now would be great

**Comment by Abukii** [comment]
Hey, is there any news about this issue? BQs TPS drops are really problematic for communities. Would love to hear good news

**Comment by Funwayguy** [comment]
Yes and no. I have been working on networking code but it's such a massive task that it's still going to be a while before any of that is ready for release. Currently build 300 has some means of preventing the serverside TPS hitching but it's no less impactful on metwork bandwidth.

**Comment by 123Kappa** [comment]
If you want some "encouragement" so you can give it more hours, I am sure that there is a lot of server owners who could gather some for you

**Comment by 123Kappa** [comment]
I can make this happen, it's a very important update for us and it has been months

**Comment by Funwayguy** [comment]
It's not so much about info. Put simply I'm having having to rewrite large portions of code to to run at the scale of an MMO without breaking existing modpacks. 

PS: I do have a Discord with Patreon channels now if you really want but that's up to you.

**Comment by 123Kappa** [comment]
Is there any ETA? Because most of the servers have started wiping maps, also if I deleted the quest file completely so it clears the player completions would that fix the issue temporary?

**Comment by Funwayguy** [comment]
No idea why they'd wipe maps, that probably wouldn't fix much of anything at all in the long run besides culling some progression data from user who no longer play (which you could do without the map erasure).

I'd suggest joining the Discord so I can give you a better rundown of where I'm at with the rewrite and what's left to do. Faster for me to answer questions about it over there too.

Link: https://discord.gg/AevsYGa
